examples/drawing/plot_rainbow_coloring.py

```diff
@@ -0,0 +1,58 @@
+"""
+================
+Rainbow Coloring
+================
+
+Generate a complete graph with `n` nodes in a circular layout with the
+edges colored by node distance. The node distance is given by the minimum
+number of nodes traversed along an arc between any two nodes on the circle.
+"""
+import matplotlib.pyplot as plt
+import networkx as nx
+
+# A rainbow color mapping using matplotlib's tableau colors
+node_dist_to_color = {
+    1: "tab:red",
+    2: "tab:orange",
+    3: "tab:olive",
+    4: "tab:green",
+    5: "tab:blue",
+    6: "tab:purple",
+}
+
+# Create a complete graph with an odd number of nodes
+nnodes = 13
+G = nx.complete_graph(nnodes)
+
+# A graph with (2n + 1) nodes requires n colors for the edges
+n = (nnodes - 1) // 2
+ndist_iter = list(range(1, n + 1))
+
+# Take advantage of circular symmetry in determining node distances
+ndist_iter += ndist_iter[::-1]
+
+
+def cycle(l, n):
+    return l[-n:] + l[:-n]
+
+
+# Rotate nodes around the circle and assign colors for each edge based on
+# node distance
+nodes = list(G.nodes())
+for i, nd in enumerate(ndist_iter):
+    for u, v in zip(nodes, cycle(nodes, i + 1)):
+        G[u][v]["color"] = node_dist_to_color[nd]
```

Comment:
It would be cool if you explicitly constructed and rotated a spanning tree to more clearly illustrate the conjecture. Perhaps that's not what this example is going for, so maybe this is good enough, but I found the article interesting.

Reply:
There is an idea floating around of repurposing the `networkx/notebooks` repo as a sort of long-form, narrative documentation/tutorials repo. One of the things we could do there is expand on the gallery examples. That's what I had in mind here - create a short example with a nice, catchy visualization - then expand upon it in a longer-form document similar to the linked article.

Reply:
Makes sense. The primary purpose seemed to just be putting up a pretty figure (which this does nicely). I figured I was conflating the purpose of the article and the purpose of this PR, but its good to know about `networkx/notebooks`. The [example I wrote](https://github.com/networkx/networkx/blob/a5c3299bf49f5823bdbb06e66766fd199a78312f/examples/applications/filesystem_embedding.py) for #4169 might make make more sense as a long-form notebook.